examples/run_field_dep_justMw.py

```diff
@@ -1,151 +1,191 @@
-date = datetime.now().strftime('%y%m%d')
 if phase_cycling:
     nPhaseSteps = 4
 if not phase_cycling:
     nPhaseSteps = 1
-p90 = 1.781
-repetition = 12e6
-SW_kHz = 10
-acq_ms = 200.
-nPoints = int(acq_ms*SW_kHz+0.5)
-tau = 3500
-#}}}
-#{{{check for file
-myfilename = date + '_'+output_name+'.h5'
-if os.path.exists(myfilename):
-    raise ValueError(
-            "the file %s already exists, change your output name!"%myfilename)
-#}}}    
-#{{{ Parameters for Bridge12
-powers = r_[3.98]
+nPoints = int(config_dict["acq_time_ms"] * config_dict["SW_kHz"] + 0.5)
+# }}}
+# {{{ Parameters for Bridge12
+powers = r_[config_dict["max_power"]]
 min_dBm_step = 0.5
 for x in range(len(powers)):
-    print(powers)
-    dB_settings = round(10*(log10(powers[x])+3.0)/min_dBm_step)*min_dBm_step # round to nearest min_dBm_step
-print("dB_settings",dB_settings)
-print("correspond to powers in Watts",10**(dB_settings/10.-3))
+    dB_settings = (
+        round(10 * (log10(powers[x]) + 3.0) / min_dBm_step) * min_dBm_step
+    )  # round to nearest min_dBm_step
+print("dB_settings", dB_settings)
+print("correspond to powers in Watts", 10 ** (dB_settings / 10.0 - 3))
 input("Look ok?")
-powers = 1e-3*10**(dB_settings/10.)
-uw_dip_center_GHz = 9.82
-uw_dip_width_GHz = 0.02
-#}}}
-#{{{run field sweep
-total_pts = nPoints*nPhaseSteps
-assert total_pts < 2**14, "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"%total_pts
+powers = 1e-3 * 10 ** (dB_settings / 10.0)
+# }}}
+# {{{run field sweep
+total_pts = nPoints * nPhaseSteps
+assert total_pts < 2 ** 14, (
+    "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"
+    % total_pts
+)
 with power_control() as p:
-    dip_f = p.dip_lock(uw_dip_center_GHz - us_dip_width_GHz/2,
-            uw_dip_center_GHz + us_dip_width_GHz/2)
-    mw_freqs.append(dip_f)
-    p.set_power9dB_settings)
-    this_dB = dBsettings
+    dip_f = p.dip_lock(
+        config_dict["uw_dip_center_GHz"] - config_dict["uw_dip_width_GHz"] / 2,
+        config_dict["uw_dip_center_GHz"] + config_dict["uw_dip_width_GHz"] / 2,
+    )
+    config_dict["mw_freqs"] = "%.9f" % dip_f
+    p.set_power(dB_settings)
     for k in range(10):
         time.sleep(0.5)
-        if p.get_power_setting()>= this_dB: break
-    if p.get_power_setting() < this_dB: raise ValueError("After 10 tries, this power has still not settled")
+        if p.get_power_setting() >= dB_settings:
+            break
+    if p.get_power_setting() < dB_settings:
+        raise ValueError("After 10 tries, this power has still not settled")
     meter_powers = np.zeros_like(dB_settings)
     with xepr() as x_server:
         first_B0 = x_server.set_field(field_axis[0])
         time.sleep(3.0)
-        carrierFreq_MHz = gamma_eff * first_B0
+        carrierFreq_MHz = config_dict["gamma_eff_MHz_G"] * first_B0
```

Comment:
carrier frequency changes with field, we calculate the new carrier frequency using the field and gamma effective